inc/cachify.class.php

```diff
@@ -91,6 +91,8 @@ public function __construct() {
 		);
 
 		/* Flush Hooks */
+		self::register_flush_cache_hooks();
```

Comment:
This method should be called later, I would say [init](https://developer.wordpress.org/reference/hooks/init/) hook should suffice. Right now it [gets executed](https://github.com/pluginkollektiv/cachify/blob/master/cachify.php#L46-L52) in context of [plugins_loaded](https://developer.wordpress.org/reference/hooks/plugins_loaded/) hook, but this is too early for themes to hook to the new `cachify_flush_cache_hooks` filter.

So this would be right way to go:
```php
add_action( 'init', array( __CLASS__, 'register_flush_cache_hooks' ), 10, 0 );
```

Also, you need to make the method public, otherwise the callback will fail to execute.

Reply:
Maybe `plugins_loaded` is a better option, runs earlier, runs before load translations, there write inside cache?

Reply:
Let me explain with example. Let's say someone wants to not flush the cache when a post is trashed. He adds the following code in his theme's `functions.php`:

```php
add_filter( 'cachify_flush_cache_hooks', function ( $hooks ) {
    unset( $hooks[ 'wp_trash_post' ] ); // Note: assumes that $flush_cache_hooks is refactored as I suggested
    return $hooks;
} );
```

However, `Cachify::register_flush_cache_hooks()` is executed in `plugins_loaded` action, so the call to `apply_filters( 'cachify_flush_cache_hooks', ... );` happens before `functions.php` is even loaded and the filter above is never run.

That's why `Cachify::register_flush_cache_hooks()` must be postponed to `init` action (or [`after_setup_theme`](https://developer.wordpress.org/reference/hooks/after_setup_theme/) at earliest).